src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -343,3 +422,95 @@ def generate_grant_sql_ddl_statements(
         grant_sql_statements.append(grant_sql_statement)
 
     return "\n".join(grant_sql_statements)
+
+
+def generate_new_sql_file_name(parent_dir: Path, file_name: str) -> Optional[Path]:
+    """
+    Generates a SQL filename using the provided parent directory, with file_name as the new file's name.
+    """
+    sql_file_name = f"{file_name}.sql"
+    sql_file_path = Path(parent_dir, sql_file_name)
+    if sql_file_path.is_file():
+        cc.warning(
+            f"File {sql_file_path} already exists, cannot overwrite existing files."
+        )
+        cc.warning("Skipping creation of this file.")
+        return None
+    return sql_file_path
+
+
+def create_and_write_to_sql_files(
+    dest_file_py_file_to_ddl_map: Dict[Path, str],
+    deploy_root: Path,
+    generated_root: Path,
+) -> Dict[Path, Path]:
+    """
+    Creates SQL files at predetermined location as specified in the input dictionary,
+    and writes the contents from the map to the new file.
+    """
+    dest_file_to_sql_file: Dict[Path, Path] = {}
+    absolute_deploy_root = resolve_without_follow(deploy_root)
+    absolute_generated_root = resolve_without_follow(generated_root)
+
+    for dest_file, all_ddl in dest_file_py_file_to_ddl_map.items():
+        if all_ddl.strip() == "":
+            continue
+
+        absolute_dest = resolve_without_follow(dest_file)
+        relative_dest_file = absolute_dest.relative_to(absolute_deploy_root)
+
+        sql_file_path = generate_new_sql_file_name(
+            parent_dir=Path(absolute_generated_root, relative_dest_file.parent),
+            file_name=f"{dest_file.stem}",
+        )
+        if sql_file_path is None:
+            # We lose information in this step, since the file already exists and we skip the overwrite.
+            continue
+
+        sql_file_path.parent.mkdir(parents=True, exist_ok=True)
+
+        with open(sql_file_path, "w+") as file:
+            file.write(all_ddl)
```

Comment:
nit: see above, DDL is generated in pieces, it would be preferable to write it in pieces, otherwise we do a lot of work in memory for no particular reason (these files could be big)